rust/models/src/resources/check.rs

```diff
@@ -0,0 +1,70 @@
+/// Checks for relative resource availability.
+///
+/// When e.g. min_free_memory is set to 10 then 10% of the memory must be free so that `verify`
+/// does not return the memory observation as an unfullfiled check.
+#[derive(Debug)]
+pub struct Checker {
+    /// Memory in bytes that must be free
+    pub memory: Option<u64>,
+    /// Percentage of CPU until that must be free.
+    pub cpu: Option<f32>,
+}
+
+impl Checker {
+    /// Returns a instance based on given absolute memory values and relative cpu usage.
+    pub fn new(memory: Option<u64>, cpu: Option<f32>) -> Self {
+        Self { memory, cpu }
+    }
+    /// Returns a instance based on relative memory instead of absolute.
+    ///
+    /// When creating an instance with 16 GB of Ram and memory of 0.1 means that at least 1,6 GB
+    /// must be available.
+    pub fn new_relative_memory(memory: f32, cpu: Option<f32>) -> Self {
+        let memory = {
+            let system = sysinfo::System::new();
+            Some((system.total_memory() as f32 * memory) as u64)
```

Comment:
```suggestion
            let system = sysinfo::System::new();
            system.refresh_all();
            Some((system.total_memory() as f32 * memory) as u64)
```
Before refreshing the first time, all values in system are not initialized